### ISSUE TITLE
Server crashes when robot attempts to send data [master 11/9/17]

### ISSUE THREAD

**Issue by system86** [dunstad/roboserver#6]
The roboserver server crashes whenever the robot sends ANYTHING. Nothing much more to say.
Using master branch from 11/9/17 

MC 1.7.10 OC 1.6.1 (i think)

Steps to reproduce are simple, download current master branch server, and current robot script, set up as usual and observe that the server crashes whenever data is sent to it.

Browser : Firefox v. 56.0.2 64 bit

Basically all versions have crippling bugs in them (in 1.0.1, robots cant connect "robot [name] is already connected", the browser doesn't work in 1.0.0).

Pertaining error : 

SyntaxError: Unexpected end of JSON input
    at parse (<anonymous>)
    at Array.map (<anonymous>)
    at Socket.tcpSocket.on ([redacted]\roboserver-master\public\js
\server\customizeServer.js:79:47)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at TCP.onread (net.js:594:20)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! roboserver@1.0.0 server: `node ./bin/www`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the roboserver@1.0.0 server script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     [redacted]\AppData\Roaming\npm-cache\_logs\2017-11-09T21_38_2
5_627Z-debug.log

**Comment by system86** [comment]
Please test your programs before releasing them.

**Comment by dunstad** [comment]
The master branch isn't stable at present. I'm aware this is unintuitive, and will be fixing it in the relatively near future. Sorry for the inconvenience. My advice would be to use one of the [releases](https://github.com/dunstad/roboserver/releases), but it sounds like you've tried at least one of them, is that correct? I think we should focus on helping you get one specific version running first. 

By "set up as usual," am I safe assuming you followed the server setup instructions [here](https://github.com/dunstad/roboserver/blob/master/documentation/server-install.md), and the robot setup instructions either [here](https://github.com/dunstad/roboserver/blob/master/documentation/survival-robot-install.md) or [here](https://github.com/dunstad/roboserver/blob/master/documentation/creative-robot-install.md)?

**Comment by system86** [comment]
I've tried all 3 branches, all have crippling bugs.

**Comment by system86** [comment]
And yes, i followed those instructions.

**Comment by system86** [comment]
& soz for being an dick in general

**Comment by dunstad** [comment]
I'll attempt to reproduce your issues by doing a fresh install of the 1.0.1 version on my own system. I'll let you know how it goes! (It'll be a bit until I can though)

**Comment by dunstad** [comment]
My best guess as to how this happened is I created 1.0.1 while the repository had a different set of robot code than I was using to test it. I marked it as a pre-release for the reason that I didn't test it as much, and I suppose this is the result. You're right that 1.0 also has significant issues. Eventually I'll get around to releasing a new version that solves these problems, and I'll close this issue then.
